src/SystemState.ts

```diff
@@ -22,6 +22,18 @@ export class SystemState {
         this._state = ndarray(new Array<number>(len).fill(0), [nGroups, nParticles, nStateElements]);
     }
 
+    public get nGroups(): number {
+        return this._nGroups;
+    }
+
+    public get nParticles(): number {
+        return this._nParticles;
+    }
+
+    public get nStateElements(): number {
```

Comment:
in the dust code this is `nState` (or really `n_state` of course), but your version is more explicit which is fine if you prefer